example/example

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+set -e
+
+NETWORK=example
+PROXY_BRANCH=mrc-3897
+
+HERE=$(realpath $(dirname $0))
+
+echo "Runing in $HERE"
+
+if [[ ! -d $HERE/config/demo ]]; then
+    mkdir -p $HERE/config
+    git clone https://github.com/mrc-ide/wodin-demo-config config/demo
+fi
+
+docker pull mrcide/wodin:mrc-3896
+docker pull mrcide/wodin-proxy:$PROXY_BRANCH
+docker pull mrcide/odin.api:main
+
+function cleanup() {
+  set +e
+  docker kill wodin-proxy
+  docker kill wodin-demo
+  docker kill odin.api
+  docker kill wodin-redis
+  docker rm wodin-proxy wodin-demo odin.api wodin-redis
+}
+trap cleanup EXIT
+
+docker rm --force wodin-redis odin.api wodin-demo wodin-proxy 2> /dev/null || \
+    /bin/true
+docker network create $NETWORK 2> /dev/null || /bin/true
+
+# Shared:
+docker run  -d --name wodin-redis --network=$NETWORK redis:6
+docker run  -d --name odin.api --network=$NETWORK mrcide/odin.api:main
+
+# We need something more sophisticated here, but we require that redis
+# is fully up and accepting connections before we try to start wodin,
+# or it will just fail to come up at all.
+echo "Waiting for redis to become available..."
+sleep 1
+
+docker run  -d --name wodin-demo --network=$NETWORK \
+       -v $HERE/config/demo:/wodin/config:ro \
+       mrcide/wodin:mrc-3896 --redis-url=redis://wodin-redis:6379 --base-url=http://localhost/demo /wodin/config
+
+# We need something more sophisticated here too, or the proxy does not
+# come up.
+echo "Waiting for wodin to become available..."
+sleep 1
```

Comment:
Should it ping the port in the container until it gets a successful response?

Reply:
yeah, but this is actually quite hard to do in a bash script - the example script here is not really designed to be used in anger. The deploy tool will do exactly what you suggest - we'll repeatedly request the wodin root endpoint until success